crates/solvers/src/boundary/baseline.rs

```diff
@@ -54,23 +60,47 @@ impl<'a> Solver<'a> {
                             path,
                             &self.amms,
                         )?;
-                        let segments = self.traverse_path(
-                            &estimate.path,
-                            request.sell.token.0,
-                            estimate.value,
-                        )?;
 
-                        let sell = estimate.value;
-                        let buy = segments
-                            .last()
-                            .map(|segment| segment.output.amount)
-                            .unwrap_or(sell);
-
-                        if sell <= request.sell.amount && buy >= request.buy.amount {
-                            Some((segments, sell))
-                        } else {
-                            None
+                        // Some baseline liquidity is "unstable", where if you
+                        // compute an input amount large enough to buy X tokens,
+                        // selling the computed amount over the same pool in the
+                        // exact same state will yield X-𝛿 tokens. To work
+                        // around this, try to converge to some `sell` amount
+                        // that produces enough `buy` tokens for the order.
+                        const MAX_ITERATIONS: usize = 3;
+                        let mut sell = estimate.value;
+                        for _ in 0..MAX_ITERATIONS {
+                            if sell > request.sell.amount {
+                                break;
+                            }
+
+                            let Some(segments) =
+                                self.traverse_path(&estimate.path, request.sell.token.0, sell)
+                            else {
+                                continue;
+                            };
+
+                            let buy = segments
+                                .last()
+                                .map(|segment| segment.output.amount)
+                                .unwrap_or(sell);
+                            if buy >= request.buy.amount {
+                                return Some((segments, sell));
+                            }
+
+                            // The computed output amount is not enough for the
+                            // order, so scale the sell amount up a bit.
+                            let bump = cmp::max(
+                                (request.buy.amount - buy)
+                                    .checked_mul(sell)?
+                                    .checked_ceil_div(&buy)?
+                                    .checked_mul(2.into())?,
```

Comment:
I initially thought this doing a binary search but it's actually doing this, right?
```
bump = convert_to_sell(missing_buy_amount) * 2
```
Am I getting this right? Seems surprising to me.

Reply:
I changed it to `bump = convert_to_sell(missing_buy_amount)` - the *2 is confusing because it wasn't actually needed 🫣.

Reply:
> doing something like a binary search

We would need to know a reasonable upper bound, which we don't. Instead I'm linearly interpolating what the sell amount should be, so something like:

```
bump = convert_to_sell(missing_buy_amount)
```